File: code/Class.m
// Basic
#import <Foundation/Foundation.h>

@interface MyClass : NSObject
#define EASY_CONSTANT @"easy_constant"
extern NSString * const MYCONSTANT;

@property(nonatomic, strong) NSString *myProperty;

-(NSString *) myMethod:(NSString *)myParam;
-(void) myOtherMethod;
@end


@implementation MyClass
NSString * const MYCONSTANT = @"constant";

// is a class method, run before any instances of the class are created and
// before other class methods are run
+(void) initialize {
}

// designated initializer
-(instancetype) init {
    if (self = [super init]) {
        self.myProperty = @"";
    }
    return self;
}

-(NSString *) myMethod:(NSString *)myParam {
    return [NSString stringWithFormat:@"in myMethod with param %@", myParam];
}

-(void) myOtherMethod {
    NSLog(@"in myOtherMethod");
}
@end

File: code/Class.swift
// Basic
import UIKit

class myClass: NSObject {
    var        myProperty:String
    let        MYCONSTANT     = "constant"
    static var staticVariable = "value"

    // Static method
    static func baz () {
      print("static function")
    }

    class func myMethodClass() {
        print("method class")
    }

    override init() {
      self.myProperty = ""
    }

    // Invocation call:
    // let foo = myClass("ABC")
    override init(_ value:String) {
      self.myProperty = value
    }

    // Invocation call:
    // let foo = myClass(value: "ABC")
    override init(value: String) {
      self.myProperty = value
    }

    deinit {
    }

    func myFirstMethod() {
        print("in myOtherMethod")
    }

    func mySecordMethod(value: String) -> String {
        return " param \(value)"
    }

    func myThirdMethod(first value1: String, second value2:String) -> String {
        return " param one \(value1) and two \(value2)"
    }

}
